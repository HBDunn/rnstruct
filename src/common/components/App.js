/**
* https://github.com/facebook/react-native
* @lint-ignore-every XPLATJSCOPYRIGHT1
*/
'use strict';

import Render from './AppRender';

import { Component } from 'react';

export default class App extends Component {
  render () {
    return Render.call(this, this.props, this.state);
  }
}
