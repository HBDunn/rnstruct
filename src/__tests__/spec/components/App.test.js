'use strict';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../../common/components/App.js';

const getbaseChild = (obj) => {
  let objs = []
  if (obj  && {}.hasOwnProperty.call(obj,"children")){
      objs = [...objs, getbaseChild(obj.children)];
  } else if (obj && Array.isArray(obj)){
    for(let el of obj){
      objs = [...objs,getbaseChild(el)];
    }
  } else if (obj) {
    objs = [...objs, obj];
  }
  return objs
};
const hasElement = (tree,str) => {
  //reduce  objs [ [ [ [Array] ], [ [Array] ], [ [Array] ] ] ]
  return getbaseChild(tree).join(",").indexOf(str) != -1;
}

  it('should have a "0" in display', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(hasElement(tree,"0")).toBe(true);
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
