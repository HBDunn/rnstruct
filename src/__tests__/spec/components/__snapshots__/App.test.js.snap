// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="main"
>
  <div
    className="screen"
  >
    0
  </div>
  <div
    className="formulae"
  />
  <div
    className="keyboard"
  >
    <div
      className="keyboard-row"
    >
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          1
        </div>
      </div>
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          2
        </div>
      </div>
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          3
        </div>
      </div>
    </div>
    <div
      className="keyboard-row"
    >
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          4
        </div>
      </div>
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          5
        </div>
      </div>
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          6
        </div>
      </div>
    </div>
    <div
      className="keyboard-row"
    >
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          7
        </div>
      </div>
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          8
        </div>
      </div>
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          9
        </div>
      </div>
    </div>
    <div
      className="keyboard-row"
    >
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          +/-
        </div>
      </div>
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          0
        </div>
      </div>
      <div
        className="key key-number"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className=""
        >
          .
        </div>
      </div>
    </div>
    <div
      className="keyboard-row"
    >
      <div
        className="key key-operator"
      >
        <div
          className="operator divide"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          ➗
        </div>
      </div>
      <div
        className="key key-operator"
      >
        <div
          className="operator substract"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          ➖
        </div>
      </div>
      <div
        className="key key-operator"
      >
        <div
          className="operator add"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          ➕
        </div>
      </div>
      <div
        className="key key-operator"
      >
        <div
          className="operator multiply"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          ✖
        </div>
      </div>
    </div>
    <div
      className="keyboard-row"
    >
      <div
        className="key key-action"
      >
        <div
          className="action back"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          &lt;&lt;
        </div>
      </div>
      <div
        className="key key-action"
      >
        <div
          className="action equal"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          =
        </div>
      </div>
    </div>
  </div>
</div>
`;
